users/api/transport.go

```diff
@@ -41,11 +41,20 @@ func MakeHandler(svc users.Service, tracer opentracing.Tracer) http.Handler {
 
 	mux.Post("/users", kithttp.NewServer(
 		kitot.TraceServer(tracer, "register")(registrationEndpoint(svc)),
-		decodeCredentials,
+		decodeCreateUserReq,
 		encodeResponse,
 		opts...,
 	))
 
+	if v := mainflux.Env("MF_SELFREGISTER", ""); v != "" {
```

Comment:
I think that for the sake of clean code reading of environment variable should be in the same place as for other variables ( users/main.go ), than you pass that to `users := service.New(...., selfergister )` and in `SelfRegister` in `service.go:users.go` you proceed or return an message if `selfregister = false`, @dusanb94 , what do you think

Reply:
`SelfRegister` in `users/service.go` is also used for admin creation in [`cmd/users/main.go`](https://github.com/mainflux/mainflux/blob/e3f90ee936d0ada0b7ed392203bf86685e6d93b5/cmd/users/main.go#L334). In order to create an admin, `SelfRegister` should be enabled. If `SelfRegister` is disabled via the `.env` file, admin creation will be failed. So, in my opinion, `SelfRegister` as a method of `users.Service` should be allowed to call but serving it through HTTP endpoint should not. 

Maybe, I can read the environment variable in the `cmd/user/main.go` and pass the environment variable to [api.MakeHandler](https://github.com/mainflux/mainflux/blob/e3f90ee936d0ada0b7ed392203bf86685e6d93b5/users/api/transport.go#L35)